### ISSUE TITLE
Panic on removing a node

### ISSUE THREAD

**Issue by tombee** [docker/docker#24227]
**Output of `docker version`:**

```
docker version
Client:
 Version:      1.12.0-rc2
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   906eacd
 Built:        Fri Jun 17 20:45:29 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-rc2
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   906eacd
 Built:        Fri Jun 17 20:45:29 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.0-rc2
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: pending
 NodeID: bytrwt3fgpugy864asuez0uf2
 IsManager: No
Runtimes: default
Default Runtime: default
Security Options: seccomp
Kernel Version: 4.4.13-boot2docker
Operating System: Boot2Docker 1.12.0-rc2 (TCL 7.1); HEAD : 52952ef - Fri Jun 17 21:01:09 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 995.9 MiB
Name: node-1
ID: N6NB:64P6:AUDY:JRUL:PJ7B:J5SN:HUHP:7V2C:LV3C:7W7G:YATG:72QO
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 24
 Goroutines: 44
 System Time: 2016-07-01T16:39:59.319294182Z
 EventsListeners: 0
Username: tombee
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

1.12-rc2 binary running against a 1.12-rc2 boot2docker VM using docker-machine.

**Steps to reproduce the issue:**
1. Disable auto-accept on swarm
2. `docker swarm join   --ca-hash sha256:b41fe431e0c28a53ba65e65d8f2076b75110b4846d327482e5fb703a741ecba1   192.168.99.100:2377`
3. `HTTP DELETE` to `/nodes/<id>`

**Describe the results you received:**

```
time="2016-07-01T15:45:36.181763764Z" level=debug msg="Calling DELETE /nodes/bytrwt3fgpugy864asuez0uf2" 
time="2016-07-01T15:45:36.183993435Z" level=debug msg="ReleaseAddress(GlobalDefault/10.255.0.0/16, 10.255.0.9)" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x58 pc=0x123f668]
goroutine 594 [running]:
panic(0x1ac37e0, 0xc82000e100)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/swarmkit/manager/scheduler.(*ReadyFilter).Check(0x2b16308, 0xc820d28bc0, 0xc8209acf01)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/scheduler/filter.go:32 +0x8
github.com/docker/swarmkit/manager/scheduler.(*Pipeline).Process(0xc8215bac40, 0xc820d28bc0, 0xc821036920)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/scheduler/pipeline.go:40 +0xc7
github.com/docker/swarmkit/manager/scheduler.(*Scheduler).taskFitNode(0xc8216a8660, 0x7f175ac598a8, 0xc82129dc40, 0xc821bd5320, 0xc820d8c480, 0x19, 0x0)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/scheduler/scheduler.go:373 +0x1af
github.com/docker/swarmkit/manager/scheduler.(*Scheduler).processPreassignedTasks(0xc8216a8660, 0x7f175ac598a8, 0xc82129dc40)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/scheduler/scheduler.go:264 +0x1a2
github.com/docker/swarmkit/manager/scheduler.(*Scheduler).Run(0xc8216a8660, 0x7f175ac598a8, 0xc82129dc40, 0x0, 0x0)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/scheduler/scheduler.go:135 +0x3a7
github.com/docker/swarmkit/manager.(*Manager).Run.func2.6(0x7f175ac598a8, 0xc82129dc40, 0xc8216a8660)
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/manager.go:377 +0x4a
created by github.com/docker/swarmkit/manager.(*Manager).Run.func2
    /go/src/github.com/docker/docker/vendor/src/github.com/docker/swarmkit/manager/manager.go:380 +0x1415
```

**Comment by LK4D4** [comment]
ping @aaronlehmann

**Comment by tombee** [comment]
ping @thaJeztah @stevvooe

**Comment by aaronlehmann** [comment]
Opened https://github.com/docker/swarmkit/pull/1115

**Comment by thaJeztah** [comment]
thanks!

**Comment by aaronlehmann** [comment]
This can be closed the next time swarmkit is revendored.

**Comment by tiborvass** [comment]
Fixed by #23312
